mlflow/server/handlers.py

```diff
@@ -1000,6 +1000,63 @@ def _get_metric_history():
     return response
 
 
+@catch_mlflow_exception
+@_disable_if_artifacts_only
+def get_metric_history_bulk_handler():
+    MAX_HISTORY_RESULTS = 25000
+    MAX_RUN_IDS_PER_REQUEST = 20
+    request_params = request.args.to_dict(flat=False)
+    run_ids = request_params.get("run_id", [])
+    if not run_ids:
+        raise MlflowException(
+            message="GetMetricHistoryBulk request must specify at least one run_id.",
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+    if len(run_ids) > MAX_RUN_IDS_PER_REQUEST:
+        raise MlflowException(
+            message=(
+                f"GetMetricHistoryBulk request cannot specify more than {MAX_RUN_IDS_PER_REQUEST}"
+                f" run_ids. Received {len(run_ids)} run_ids."
+            ),
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+
+    metric_key = request_params.get("metric_key", [])
+    if not metric_key:
+        raise MlflowException(
+            message="GetMetricHistoryBulk request must specify a metric_key.",
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+    metric_key = metric_key[0]
+
+    max_results = int(request_params.get("max_results", [MAX_HISTORY_RESULTS])[0])
+    max_results = min(max_results, MAX_HISTORY_RESULTS)
+
+    store = _get_tracking_store()
+    metrics_with_runids = []
```

Comment:
```suggestion
    metrics_with_run_ids = []
```

Reply:
Applied this suggestion! Thanks @apurva-koti !